src/vulkan-renderer/wrapper/device.cpp

```diff
@@ -311,74 +311,88 @@ Device::~Device() {
 }
 
 #ifndef NDEBUG
-void Device::set_object_name(const std::uint64_t object, const VkDebugReportObjectTypeEXT type,
-                             const std::string &name) {
-    assert(m_device);
-    assert(!name.empty());
-    assert(object);
-    assert(m_vk_debug_marker_set_object_name);
+void Device::set_object_name(const void *object, const VkDebugReportObjectTypeEXT type, const std::string &name) const {
```

Comment:
Why don't we use `std::uint64_t` instead of `const void*`? and reinterpretated while passing the argument.
Just an idea.

Reply:
Haha, I actually told @IAmNotHanni to switch the argument from `std::uint64_t` to `void *` in my review, because I felt like doing the cast once would be nicer than doing it in every other wrapper.

Reply:
Iam unsure myself, what would be better.
On one hand there is the better signature on the other hand you have to use reinterpretate cast.

Reply:
And a void pointer is more or less always be, what exactly do i have to pass now.

Reply:
Oh right, I think that would make it more explicit + type safe?

Reply:
Done.